string[] var_names
string[] types
string[] values
---
string message